### DISCUSSION TITLE
how to seprate column def

### DISCUSSION THREAD

**Discussion by 0xMALVEE** [TanStack/table#5210] [Q&A]
![image](https://github.com/TanStack/table/assets/35369563/d8306f96-b359-4a66-90b2-db3e48bd81ba)
I want moved my column def in another file but since it's not a react component ig not working can't use React.useMemo directly in here. so any way to seperate in another file so all working? if I remove useMemo there will be any iisues?

**Comment by ronaldruzicka** [comment]
I guess you can create a hook, that returns `columnTraderStrategies`, then you can use `useMemo` inside
```ts
export function useTraderStrategiesColumns() {
  const traderStrategiesColumns = useMemo(...);

  return traderStrategiesColumns;
}
```
